csharp/method.go

```diff
@@ -145,6 +160,10 @@ func (self *MethodDeclaration) WriteCode(writer CodeWriter) {
 			}
 		}
 		writer.Write(")")
+
+		if self.hasBase && self.base != nil {
```

Comment:
you could probably just use the nil check instead of manually keeping `hasBase`

Reply:
i was trying to follow the convention but true indeed.

Reply:
actually, i prefer to keep it as it prevents someone from generating a regular method declaration with the ctor base call syntax.

Reply:
> actually, i prefer to keep it as it prevents someone from generating a regular method declaration with the ctor base call syntax.

not sure Im following, you mean if someone renders a `BaseStatement` outside the context of a method?